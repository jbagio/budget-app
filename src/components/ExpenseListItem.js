import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ id, description, amount, createdAt }) => {
  return (
    <div>
      <h3>
        <Link to={`/edit/${id}`}>{description}</Link>
      </h3>
      <p>
        {amount} - {createdAt}
      </p>
    </div>
  );
};

ExpenseListItem.propTypes = {
  id: PropTypes.string,
  description: PropTypes.string,
  amount: PropTypes.number,
  createdAt: PropTypes.number,
};

export default ExpenseListItem;
