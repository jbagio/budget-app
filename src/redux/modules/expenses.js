import uuid from 'uuid';

const ADD_EXPENSE = 'budget-app/expenses/add-expense';
const REMOVE_EXPENSE = 'budget-app/expenses/remove-expense';
const EDIT_EXPENSE = 'budget-app/expenses/edit-expense';

export const addExpense = ({
  description = '',
  note = '',
  amount = 0,
  createdAt = 0,
} = {}) => ({
  type: ADD_EXPENSE,
  expense: {
    id: uuid(),
    description,
    note,
    amount,
    createdAt,
  },
});

export const removeExpense = (id = {}) => ({
  type: REMOVE_EXPENSE,
  id,
});

export const editExpense = (id, updatedExpense) => ({
  type: EDIT_EXPENSE,
  id,
  updatedExpense,
});

const initialState = [];

export default function reducer(state = initialState, action) {
  switch (action.type) {
    case ADD_EXPENSE:
      return [...state, action.expense];
    case REMOVE_EXPENSE:
      return state.filter(expense => expense.id !== action.id);
    case EDIT_EXPENSE:
      return state.map(expense => {
        if (expense.id === action.id) {
          return {
            ...expense,
            ...action.updatedExpense,
          };
        }
        return expense;
      });
    default:
      return state;
  }
}
