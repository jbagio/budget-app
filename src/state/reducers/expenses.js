import * as types from '../actions/actionTypes';

const initialState = [];

export default (state = initialState, action) => {
  switch (action.type) {
    case types.ADD_EXPENSE:
      return [...state, action.expense];
    case types.REMOVE_EXPENSE:
      return state.filter(expense => expense.id !== action.id);
    case types.EDIT_EXPENSE:
      return state.map(expense => {
        if (expense.id === action.id) {
          return {
            ...expense,
            ...action.updatedExpense,
          };
        }
        return expense;
      });
    default:
      return state;
  }
};
