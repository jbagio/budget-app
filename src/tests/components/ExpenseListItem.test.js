import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

it('should render ExpenseListItem correctly', () => {
  const wrapper = shallow(<ExpenseListItem expense={expenses[1]} />);

  expect(wrapper).toMatchSnapshot();
});
