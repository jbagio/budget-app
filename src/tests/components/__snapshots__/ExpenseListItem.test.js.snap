// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem correctly 1`] = `
<div>
  <h3>
    <Link
      to="/edit/undefined"
    />
  </h3>
  <p>
     - 
  </p>
</div>
`;
